azure-ip-masq-merger/ip_masq_merger.go

```diff
@@ -0,0 +1,276 @@
+package main
+
+import (
+	utiljson "encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/fs"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/component-base/logs"
+	"k8s.io/component-base/version/verflag"
+	"k8s.io/klog/v2"
+)
+
+// Version is populated by make during build.
+var version string
+
+const (
+	// path to a yaml or json files
+	configPath = "/etc/config/"
+	// merged config written to this directory
+	outputPath = "/etc/merged-config/"
+	// config files in this path must start with this to be read
+	configFilePrefix = "ip-masq"
+)
+
+type FileSystem interface {
+	ReadFile(name string) ([]byte, error)
+	WriteFile(name string, data []byte, perm fs.FileMode) error
+	ReadDir(dirname string) ([]fs.DirEntry, error)
+	DeleteFile(name string) error
+}
+
+type OSFileSystem struct{}
+
+func (OSFileSystem) ReadFile(name string) ([]byte, error) {
+	return os.ReadFile(name) // nolint
+}
+
+func (OSFileSystem) WriteFile(name string, data []byte, perm fs.FileMode) error {
+	return os.WriteFile(name, data, perm) // nolint
+}
+
+func (OSFileSystem) ReadDir(dirname string) ([]fs.DirEntry, error) {
+	return os.ReadDir(dirname) // nolint
+}
+
+func (OSFileSystem) DeleteFile(name string) error {
+	return os.Remove(name) // nolint
+}
+
+// name of nat chain for iptables masquerade rules
+var resyncInterval = flag.Int("resync-interval", 60, "How often to refresh the config (in seconds)")
+
+// MasqConfig object
+type MasqConfig struct {
```

Comment:
Reuse https://github.com/kubernetes-sigs/ip-masq-agent/blob/master/cmd/ip-masq-agent/ip-masq-agent.go#L68?

Reply:
If we include the resync interval and cidr limit from upstream, how would this be merged? I initially started with the one from upstream as you mentioned but ran into the above issue.